cid/common.py

```diff
@@ -1174,13 +1229,49 @@ def create_or_update_dataset(self, dataset_definition: dict, dataset_id: str=Non
 
         found_dataset = self.qs.describe_dataset(compiled_dataset.get('DataSetId'))
         if isinstance(found_dataset, Dataset):
+            update_dataset = False
             if update:
-                self.qs.update_dataset(compiled_dataset)
+                update_dataset = True
             elif found_dataset.name != compiled_dataset.get('Name'):
                 print(f"Dataset found with name {found_dataset.name}, but {compiled_dataset.get('Name')} expected. Updating.")
+                update_dataset = True
+            if update_dataset and get_parameters().get('on-drift', 'show').lower() != 'override' and isatty() and not cur_required:
+                while True:
+                    diff = self.qs.dataset_diff(found_dataset.raw, compiled_dataset)
+                    if diff and diff['diff']:
+                        cid_print(f'<BOLD>Found a difference between existing dataset <YELLOW>{found_dataset.name}<END> <BOLD>and the one we want to deploy. <END>')
+                        cid_print(diff['printable'])
+                        choice = get_parameter(
+                            param_name='dataset-' + found_dataset.name.lower().replace(' ', '-') + '-override',
+                            message=f'The existing dataset is different. Override?',
+                            choices=['retry diff', 'proceed and override', 'keep existing', 'stop'],
+                            default='retry diff'
+                        )
+                        if choice == 'retry diff':
+                            unset_parameter('dataset-' + found_dataset.name.lower().replace(' ', '-') + '-override')
+                            continue
+                        elif choice == 'proceed and override':
+                            update_dataset = True
+                            break
+                        elif choice == 'keep existing':
+                            update_dataset = False
+                            break
+                        else:
+                            raise CidCritical(f'User choice is not to update {found_dataset.name}.')
+                    elif not diff:
+                        if not get_parameter(
+                            param_name=found_dataset.name.lower().replace(' ', '-') + '-override',
```

Comment:
is there any way to get rid of this replace?

Reply:
name can contain spaces. parameter names cannot. i do not see how we can workaround this without a replace